File: ios/FoodTracking/FoodTracking/Models/FavoriteFood.swift
// FavoriteFood Model - SwiftData equivalent of Room entity

import Foundation
import SwiftData

@Model
final class FavoriteFood {
    var id: UUID
    var name: String
    var calories: Int
    var fats: Float
    var proteins: Float
    var carbs: Float
    var emoji: String
    var weightGrams: Int
    var createdAt: Date
    
    init(
        id: UUID = UUID(),
        name: String,
        calories: Int,
        fats: Float = 0,
        proteins: Float = 0,
        carbs: Float = 0,
        emoji: String = "🍽️",
        weightGrams: Int = 100
    ) {
        self.id = id
        self.name = name
        self.calories = calories
        self.fats = fats
        self.proteins = proteins
        self.carbs = carbs
        self.emoji = emoji
        self.weightGrams = weightGrams
        self.createdAt = Date()
    }
    
    /// Convert to FoodEntry for today
    func toFoodEntry() -> FoodEntry {
        FoodEntry(
            name: name,
            calories: calories,
            fats: fats,
            proteins: proteins,
            carbs: carbs,
            emoji: emoji,
            weightGrams: weightGrams,
            date: Date(),
            source: .favorites
        )
    }
}

File: app/src/main/java/com/example/foodtracking/data/model/FavoriteFood.kt
package com.example.foodtracking.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * Represents a favorite food item saved by the user.
 */
@Entity(tableName = "favorite_foods")
data class FavoriteFood(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val name: String,
    val calories: Int,
    val fats: Float,
    val proteins: Float,
    val carbs: Float,
    val emoji: String = "🍽️",
    val weightGrams: Int = 100 // Default weight
) {
    fun toFoodEntry(date: java.time.LocalDate, timestamp: java.time.LocalDateTime = java.time.LocalDateTime.now()): FoodEntry {
        return FoodEntry(
            name = name,
            calories = calories,
            fats = fats,
            proteins = proteins,
            carbs = carbs,
            emoji = emoji,
            weightGrams = weightGrams,
            date = date,
            timestamp = timestamp,
            source = FoodSource.DATABASE
        )
    }
    
    companion object {
        fun fromFoodEntry(entry: FoodEntry): FavoriteFood {
            return FavoriteFood(
                name = entry.name,
                calories = entry.calories,
                fats = entry.fats,
                proteins = entry.proteins,
                carbs = entry.carbs,
                emoji = entry.emoji,
                weightGrams = entry.weightGrams
            )
        }
    }
}
